src/Voron/Impl/Paging/AbstractPager.cs

```diff
@@ -303,14 +303,21 @@ public int NodeMaxSize
                     goto AlreadyDisposed;
             }
 
-            tx?.EnsurePagerStateReference(state);
+            tx?.EnsurePagerStateReference(ref state);
+            
+            if (state._released)
+                goto InvalidPagerState;
```

Comment:
@redknightlois Please have a look at this change. I followed the way we throw exceptions in this method and used `goto` statement as in other two places here.